pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,233 @@
+// limitations under the License.
+
+package kanctl
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+)
+
+const (
+	tlsSecretFlag                       = "tlsSecret"
+	repoServerUserFlag                  = "repositoryServerUser"
+	repoServerUserAccessSecretFlag      = "repositoryServerUserAccessSecret"
+	repoServerAdminUserAccessSecretFlag = "repositoryServerAdminUserAccessSecret"
+	repoPasswordSecretFlag              = "repositoryPasswordSecret"
+	repoUserFlag                        = "repositoryUser"
+	locationCredsSecretFlag             = "locationCredsSecret"
+	locationSecretFlag                  = "locationSecret"
+	defaultKanisterNamespace            = "kanister"
+	defaultRepositoryServerHost         = "localhost"
+)
+
+type repositoryServerParams struct {
+	tls                             string
+	repositoryServerUser            string
+	repositoryServerUserAccess      string
+	repositoryServerAdminUserAccess string
+	repositoryUser                  string
+	repositoryPassword              string
+	prefix                          string
+	location                        string
+	locationCreds                   string
+}
+
+func newRepositoryServerCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "repository-server",
+		Short: "Create a new RepositoryServer",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createNewRepositoryServer(cmd, args)
+		},
+	}
+
+	cmd.PersistentFlags().StringP(tlsSecretFlag, "t", "", "name of the tls secret needed for secure kopia client and kopia repository server communication")
+	cmd.PersistentFlags().StringP(repoServerUserFlag, "u", "", "name of the user to be created for the kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerUserAccessSecretFlag, "k", "", "name of the secret having access credentials of the users that can connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerAdminUserAccessSecretFlag, "a", "", "name of the secret having admin credentials to connect to connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoPasswordSecretFlag, "r", "", "name of the secret containing password for the kopia repository")
+	cmd.PersistentFlags().StringP(prefixFlag, "p", "", "prefix to be set in kopia repository")
+	cmd.PersistentFlags().StringP(repoUserFlag, "z", "", "name of the user for accessing the kopia repository")
+	cmd.PersistentFlags().StringP(locationSecretFlag, "l", "", "name of the secret containing kopia repository storage location details")
+	cmd.PersistentFlags().StringP(locationCredsSecretFlag, "c", "", "name of the secret containing kopia repository storage credentials")
+
+	_ = cmd.MarkFlagRequired(tlsSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserFlag)
```

Comment:
Should we either handle these errors or just keep it as
```
cmd.MarkFlagRequired(repoServerUserFlag)
```

Reply:
@ankitjain235, we can leave them as it is.